indico/modules/rb/schemas.py

```diff
@@ -193,13 +196,14 @@ class Meta:
 
 class ReservationOccurrenceSchema(mm.SQLAlchemyAutoSchema):
     reservation = Nested(ReservationSchema)
+    link = Nested(ReservationOccurrenceLinkSchema)
     state = EnumField(ReservationOccurrenceState)
     start_dt = NaiveDateTime()
     end_dt = NaiveDateTime()
 
     class Meta:
         model = ReservationOccurrence
-        fields = ('start_dt', 'end_dt', 'is_valid', 'reservation', 'rejection_reason', 'state')
+        fields = ('start_dt', 'end_dt', 'is_valid', 'reservation', 'rejection_reason', 'state', 'link')
```

Comment:
This is leaking the title etc of the linked object regardless of access - for exactly this reason we lazy-load this data when viewing booking details. Maybe it'd be better to just include a flag whether the occurrence is linked to something or not? (or maybe you don't need that because `can_link` handles it?)

If one wants to know what it's linked to, they can always click on it and view the booking details.

Reply:
The only reason we need the link for is to display some details:

<img width="335" alt="Screenshot 2024-04-23 at 10 51 31" src="https://github.com/indico/indico/assets/2722756/0565fcb1-adea-4ff0-870f-06c468f7961f">

We'd loose that info if we remove that.

Reply:
How about showing something like "Already linked"? When clicking to get the usual details dialog you'll see the full link details anyway (assuming you have access).

Reply:
The `ReservationOccurrenceLinkSchema` is already taking care of this, isn't it?

![Screenshot 2024-04-23 at 11 10 23](https://github.com/indico/indico/assets/2722756/77b0080c-e353-4929-9d1e-fd01c31a3bba)

Reply:
Yes, but we want to avoid doing a huge amount of access checks on that page - the API can return a large amount of bookings, and doing an access check for so many events would quickly cause query spam, especially in a deeply nested category tree where we need to climb up the tree until we find the category that grants the user access.

Reply:
This is now changed to `link_id` instead of `link` in order not to leak info.